trillian/ctfe/ct_server/main.go

```diff
@@ -223,8 +218,21 @@ func main() {
 		}
 	}
 
-	err = http.ListenAndServe(*httpEndpoint, handler)
-	glog.Warningf("Server exited: %v", err)
+	// Bring up the HTTP server and serve until we get a signal not to.
+	srv := http.Server{Addr: *httpEndpoint, Handler: handler}
+	go awaitSignal(func() {
+		// Allow 60s for any pending requests to finish then terminate any stragglers
+		ctx, cancel := context.WithTimeout(context.Background(), time.Second*60)
+		defer cancel()
+		glog.Info("Shutting down HTTP server...")
+		srv.Shutdown(ctx)
+		glog.Info("HTTP server shutdown")
+	})
+
+	err = srv.ListenAndServe()
```

Comment:
From `Shutdown` comment:
```
// When Shutdown is called, Serve, ListenAndServe, and
// ListenAndServeTLS immediately return ErrServerClosed. Make sure the
// program doesn't exit and waits instead for Shutdown to return.
```

Should you be waiting here until the `awaitSignal` goroutine terminates?

Reply:
D'oh, yup you are correct, probably a basic wait channel would be the proper solution here... will open a fix PR.

Reply:
Yeah. Not sure about the channel though, might be easier to use a `WorkGroup` with `Add/defer Done` called inside the signal handler, and `wg.Wait` after `ListenAndServe`? Up to you though, I would be happy to review.

Reply:
#261, thanks for catching this!